src/lib.rs

```diff
@@ -735,6 +737,27 @@ pub enum BinaryOperator {
     ShiftRight,
 }
 
+/// Function on an atomic value.
+///
+/// Note: these do not include load/store, which use the existing
+/// [`Expression::Load`] and [`Statement::Store`].
+#[derive(Clone, Copy, Debug, Hash, Eq, Ord, PartialEq, PartialOrd)]
+#[cfg_attr(feature = "serialize", derive(Serialize))]
+#[cfg_attr(feature = "deserialize", derive(Deserialize))]
+pub enum AtomicFunction {
+    Binary {
+        op: BinaryOperator,
+        value: Handle<Expression>,
+    },
```

Comment:
This should document which `BinaryOperator` values are actually permitted by the validator.